### PR TITLE
Fix reverted permissions when deploying from CLI

### PR DESCRIPTION
**Pull request by colinking**
Previously, if you had explicit permissions set on a task, then deploying that task from the CLI would revert those permissions.

Resolves AIR-1556

### CODE REVIEW

File: pkg/api/types.go [hunk 1]
```diff
@@ -29,24 +29,36 @@ type CreateTaskRequest struct {
...
+	Arguments                  []string          `json:"arguments" yaml:"arguments"`
+	Parameters                 Parameters        `json:"parameters" yaml:"parameters"`
+	Constraints                RunConstraints    `json:"constraints" yaml:"constraints"`
+	Env                        TaskEnv           `json:"env" yaml:"env"`
+	ResourceRequests           map[string]string `json:"resourceRequests" yaml:"resourceRequests"`
+	Resources                  map[string]string `json:"resources" yaml:"resources"`
+	Kind                       TaskKind          `json:"kind" yaml:"builder"`
+	KindOptions                KindOptions       `json:"kindOptions" yaml:"builderConfig"`
+	Repo                       string            `json:"repo" yaml:"repo"`
+	RequireExplicitPermissions bool              `json:"requireExplicitPermissions"`
```
Comment (joshma) [inline pkg/api/types.go:46]: Missing a `yaml` tag?

File: pkg/print/types.go [hunk 2]
```diff
@@ -6,23 +6,25 @@ import (
...
+	Parameters                 api.Parameters       `json:"parameters" yaml:"parameters"`
+	Constraints                api.RunConstraints   `json:"constraints" yaml:"constraints"`
+	Env                        api.TaskEnv          `json:"env" yaml:"env"`
+	ResourceRequests           api.ResourceRequests `json:"resourceRequests" yaml:"resourceRequests"`
+	Resources                  api.Resources        `json:"resources" yaml:"resources"`
+	Kind                       api.TaskKind         `json:"builder" yaml:"builder"`
+	KindOptions                api.KindOptions      `json:"builderConfig" yaml:"builderConfig"`
+	Repo                       string               `json:"repo" yaml:"repo"`
+	RequireExplicitPermissions bool                 `json:"requireExplicitPermissions" yaml:"-"`
+	Permissions                api.Permissions      `json:"permissions" yaml:"-"`
```
Comment (joshma) [inline pkg/print/types.go:26]: If this is just for printing...can we omit it from this struct?

File: pkg/build/remote.go [hunk 3]
```diff
@@ -97,19 +97,21 @@ func updateKindAndOptions(ctx context.Context, client *api.Client, def definitio
...
+		Command:                    task.Command,
+		Arguments:                  task.Arguments,
+		Parameters:                 task.Parameters,
+		Constraints:                task.Constraints,
+		Env:                        task.Env,
+		ResourceRequests:           task.ResourceRequests,
+		Resources:                  task.Resources,
+		Repo:                       task.Repo,
+		RequireExplicitPermissions: task.RequireExplicitPermissions,
+		Permissions:                task.Permissions,
```
Comment (joshma) [inline pkg/build/remote.go:113]: Unfortunate that it has to be specified - can we not just remove this from the apiext endpoint? Is this actually used (by new or old)?

**Review by joshma — APPROVED** [review]

File: pkg/print/types.go (hunk 2 above)
Comment (colinking) [reply pkg/print/types.go:26]: We do a type conversion, so it's needed to compile. But decided not to print this metadata for now

File: pkg/build/remote.go (hunk 3 above)
Comment (colinking) [reply pkg/build/remote.go:113]: Agreed it's a pain that you have to specify fields that haven't changed. Age old problem with CRUD APIs 😄 

And it seems reasonable to me that the external API can control permissions, so I'd rather not remove it from the API.

File: pkg/print/types.go (hunk 2 above)
Comment (joshma) [reply pkg/print/types.go:26]: I think you need to do `json:"-"` too, for `-o json`?

File: pkg/build/remote.go (hunk 3 above)
Comment (joshma) [reply pkg/build/remote.go:113]: Okay, SG

**Pull request merged**